### PR TITLE
Improve null simulations and add analyses of GLS model

### PR DESCRIPTION
**Pull request by miltondp**
This PR fixes some problems with the previous null simulations for the GLS model.

The main process consists in running: GWAS, GWAS harmonization, TWAS (S-PrediXcan and S-MultiXcan) and finally the GLS model of PhenoPLIER. The folders below (which are all under `nbs/15_gsa_gls/20-null_simulations/`) contain mostly small changes with a few new files to fix the previously mentioned problems:
* `05_gwas/`
* `10_gwas_harmonization/`
* `15_spredixcan/`

The subfolder with the last step of the process (`20_gls_phenoplier`) is the one that has the main files changes that were introduced. It has scripts to run the jobs in a cluster and some notebooks to analyze the results.

### CODE REVIEW

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md [hunk 1]
```diff
@@ -135,17 +151,112 @@ mkdir -p _tmp/postprocessing
...
+input_files = sorted(list(POST_IMPUTED_DIR.glob("*.txt.gz")))
+len(input_files)
+
+# sample files
+np.random.seed(0)
+input_files = np.random.choice(input_files, size=N_SAMPLES, replace=False)
+len(input_files)
+
+# read all GWAS and find a set of common panel_variant_id_values
+def _get_gwas_variants(f):
```
Comment (rando2) [inline nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:209]: You have much more software engineering experience than I do so I'm curious --  I am not used to seeing functions embedded in the main code. When/why do you do that?

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md [hunk 2]
```diff
@@ -135,17 +151,112 @@ mkdir -p _tmp/postprocessing
...
+mkdir -p _tmp/common_var_ids
+cat cluster_jobs/15-common_variant_ids_job.sh | bsub
+```
+
+Checks:
+```bash
+bash check_job.sh \
+  -i _tmp/common_var_ids/ \
+  -f '*.out' \
+  -p "Filtering variants: 8325729"
```
Comment (rando2) [inline nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:246]: Maybe include a note about what this flag means?

**Review by rando2 — COMMENTED** [review]
Hi Milton, I just had a couple of questions about the parameterizing!

**Review by rando2 — APPROVED** [review]

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md [hunk 3]
```diff
@@ -62,9 +64,14 @@ Here we need to use some templating, because we impute across random phenotypes,
...
+# Iterate over all random phenotype ids, chromosomes and batch ids and submit a job for each combination.
+# IMPORTANT: These are a lot of tasks. You might want to split jobs by chaning the range in first for line:
+#   0..199
+#   200..399
+#   400..599
+#   600..799
+#   800..999
+for pheno_id in {0..9}; do
   for chromosome in {1..22}; do
     for batch_id in {0..9}; do
```
[the remark this answers is not in the document]
Comment (miltondp) [reply nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:76]: I've added a comment documenting this. `batch_id` is a parameter of one of the tools I'm using, basically it allows to split variants in a chromosome in more jobs and thus parallelize more.

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md [hunk 4]
```diff
@@ -62,9 +64,14 @@ Here we need to use some templating, because we impute across random phenotypes,
...
-# batch ids and submit a job for each combination
-for pheno_id in {0..99}; do
+# Iterate over all random phenotype ids, chromosomes and batch ids and submit a job for each combination.
+# IMPORTANT: These are a lot of tasks. You might want to split jobs by chaning the range in first for line:
+#   0..199
+#   200..399
+#   400..599
+#   600..799
+#   800..999
+for pheno_id in {0..9}; do
```
[the remark this answers is not in the document]
Comment (miltondp) [reply nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:74]: Yeap! Old numbers, it should be `0..999` in the documentation.

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md [hunk 5]
```diff
@@ -85,6 +95,11 @@ import os
 import re
 from pathlib import Path
 
+# adjust accordingly
+N_PHENOTYPES = 200    # 1000
```
[the remark this answers is not in the document]
Comment (miltondp) [reply nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:99]: Good catch!

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md (hunk 2 above)
Comment (miltondp) [reply nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:246]: Yes, good point. I've just added it. It's the "success pattern" that has to be found for the script to finish correctly.

File: nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md (hunk 1 above)
Comment (miltondp) [reply nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md:211]: Good question. It's not common to have a function in a bunch of code intended to be copied/pasted into an ipython console (unless it's helpful for readability). But in this case, I'm using parallelization with the `concurrent.futures` package, which needs a function that works on a single chunk of the entire data. That's why I've added this function, which is used when calling `executor.map`.

**Pull request merged**